PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/springcloud/deplolyment/SpringCloudDeploymentConfiguration.java

```diff
@@ -53,26 +52,28 @@ public SpringCloudDeploymentConfiguration(@NotNull Project project, @NotNull Con
             .build();
     }
 
-    protected SpringCloudDeploymentConfiguration(@NotNull SpringCloudDeploymentConfiguration source) {
-        super(source.getProject(), source.getFactory(), source.getName());
-        this.appConfig = JsonUtils.deepCopyWithJson(source.appConfig);
-    }
-
     @Override
     public void readExternal(Element element) throws InvalidDataException {
         super.readExternal(element);
-        XmlSerializer.deserializeInto(this.appConfig, element);
+        final Element appConfigElement = element.getChild("SpringCloudAppConfig");
+        this.appConfig = XmlSerializer.deserialize(appConfigElement, SpringCloudAppConfig.class);
+        final String artifactId = element.getChild("Artifact").getAttributeValue("identifier");
+        this.appConfig.getDeployment().setArtifact(IArtifact.fromId(artifactId));
     }
 
     @Override
     public void writeExternal(Element element) throws WriteExternalException {
         super.writeExternal(element);
-        XmlSerializer.serializeInto(this.appConfig, element, new SerializationFilterBase() {
-            @Override
-            protected boolean accepts(@NotNull Accessor accessor, @NotNull Object bean, @Nullable Object beanValue) {
-                return !"password".equalsIgnoreCase(accessor.getName());
-            }
-        });
+        final Element appConfigElement = XmlSerializer.serialize(this.appConfig, (accessor, o) -> !"artifact".equalsIgnoreCase(accessor.getName()));
+        final IArtifact artifact = this.appConfig.getDeployment().getArtifact();
+        Optional.ofNullable(this.appConfig)
+            .map(config -> XmlSerializer.serialize(config, (accessor, o) -> !"artifact".equalsIgnoreCase(accessor.getName())))
```

Comment:
StringUtils.equalsIgnoreCase

Reply:
it's safe to call `String.equalsIgnoreCase` here